planning/behavior_path_planner/src/utilities.cpp

```diff
@@ -1679,6 +1679,103 @@ PathWithLaneId getCenterLinePath(
   return route_handler.getCenterLinePath(lanelet_sequence, s_backward, s_forward, true);
 }
 
+PathWithLaneId getCenterLinePath(
+  const RouteHandler & route_handler, const lanelet::ConstLanelets & lanelet_sequence,
+  const Pose & pose, const double backward_path_length, const double forward_path_length,
+  const BehaviorPathPlannerParameters & parameter, const double & additional_length)
+{
+  PathWithLaneId reference_path;
+
+  if (lanelet_sequence.empty()) {
+    return reference_path;
+  }
+
+  const auto arc_coordinates = lanelet::utils::getArcCoordinates(lanelet_sequence, pose);
+  const double s = arc_coordinates.length;
+  const double s_backward = std::max(0., s - backward_path_length);
+  double s_forward = s + forward_path_length;
+
+  const double buffer = parameter.backward_length_buffer_for_end_of_lane +
+                        additional_length;  // buffer for min_lane_change_length
+  const int num_lane_change =
+    std::abs(route_handler.getNumLaneToPreferredLane(lanelet_sequence.back()));
+  const double lane_length = lanelet::utils::getLaneletLength2d(lanelet_sequence);
+  const double lane_change_buffer =
+    num_lane_change * (parameter.minimum_lane_change_length + buffer);
+
+  if (route_handler.isDeadEndLanelet(lanelet_sequence.back())) {
+    s_forward = std::min(s_forward, lane_length - lane_change_buffer);
+  }
+
+  if (route_handler.isInGoalRouteSection(lanelet_sequence.back())) {
+    const auto goal_arc_coordinates =
+      lanelet::utils::getArcCoordinates(lanelet_sequence, route_handler.getGoalPose());
+    s_forward = std::min(s_forward, goal_arc_coordinates.length - lane_change_buffer);
+  }
+
+  return route_handler.getCenterLinePath(lanelet_sequence, s_backward, s_forward, true);
+}
+
+bool checkLaneIsInIntersection(
+  const RouteHandler & route_handler, const PathWithLaneId & ref,
```

Comment:
Could you change the name `ref` to `reference_path` ?

Reply:
The naming is fixed now.